tests/test_scatterwidgets.py

```diff
@@ -1,58 +1,404 @@
+    plot_widget.roi_list = [roi]
+    plot_widget.plot()
+
+    assert len(plot_widget.roi_list) == 1
+
+    rois = [item for item in plot_widget.scatter_plot.items if isinstance(item, pg.RectROI)]
+
+    assert len(rois) == 1, "There should be exactly one ROI in the scatter_plot"
+
+
+def test_remove_roi_double_click(qtbot, plot_widget, prepare_disc_test_data):
+    """Test changing coordinates and removing ROI."""
+    roi = pg.RectROI([0, 0], [1, 1])
+    plot_widget._onClick(*prepare_disc_test_data)
+    plot_widget.roi_list = [roi]
+    plot_widget.plot()
+
+    assert len(plot_widget.roi_list) == 1
+
+    # Simulate double click
+    view_widget = plot_widget.scatter_plot.scene().views()[0].viewport()
+
+    point1 = [0.5, 0.5]
+    viewport_pos1 = plot_coords_to_viewport(plot_widget, *point1)
+
+    qtbot.mouseDClick(view_widget, QtCore.Qt.LeftButton, pos=viewport_pos1)
+    qtbot.wait(50)
+
+    assert len(plot_widget.roi_list) == 0
+
+
+def test_remove_hovered_roi(qtbot, plot_widget, prepare_disc_test_data):
+    """Test changing coordinates and removing ROI."""
+    roi = pg.RectROI([0, 0], [1, 1])
+    plot_widget.roi_list = [roi]
+    plot_widget.plot()
+
+    assert len(plot_widget.roi_list) == 1
+
+    # Simulate hovering over the ROI
+    roi.mouseHovering = True
+
+    plot_widget.remove_hovered_roi()
+
+    assert len(plot_widget.roi_list) == 0
+
+
+def test_remove_all_rois(qtbot, plot_widget, prepare_disc_test_data):
+    """Test changing coordinates and removing ROI."""
+    roi1 = pg.RectROI([0, 0], [0.5, 0.5])
+    roi2 = pg.RectROI([0.6, 0.6], [0.7, 0.7])
+    roi3 = pg.RectROI([0.8, 0.8], [0.9, 0.9])
+    plot_widget._onClick(*prepare_disc_test_data)
+    plot_widget.roi_list = [roi1, roi2, roi3]
+    plot_widget.plot()
+
+    assert len(plot_widget.roi_list) == 3
+
+    plot_widget.remove_all_rois()
+
+    assert len(plot_widget.roi_list) == 0
+
+
+def test_keyboard_bindings(qtbot, plot_widget, prepare_disc_test_data):
+    """Test changing coordinates and removing ROI."""
+    roi1 = pg.RectROI([0, 0], [0.5, 0.5])
+    roi2 = pg.RectROI([0.6, 0.6], [0.7, 0.7])
+    roi3 = pg.RectROI([0.8, 0.8], [0.9, 0.9])
+    plot_widget._onClick(*prepare_disc_test_data)
+    plot_widget.roi_list = [roi1, roi2, roi3]
+    plot_widget.plot()
+
+    assert len(plot_widget.roi_list) == 3
+
+    event = QtGui.QKeyEvent(QtCore.QEvent.KeyPress, QtCore.Qt.Key_E, QtCore.Qt.ShiftModifier)
+    QtWidgets.QApplication.sendEvent(plot_widget, event)
+    assert len(plot_widget.roi_list) == 3
+
+    roi2.mouseHovering = True
+    event = QtGui.QKeyEvent(QtCore.QEvent.KeyPress, QtCore.Qt.Key_D, QtCore.Qt.NoModifier)
+    QtWidgets.QApplication.sendEvent(plot_widget, event)
+    assert len(plot_widget.roi_list) == 2
+
+    event = QtGui.QKeyEvent(QtCore.QEvent.KeyPress, QtCore.Qt.Key_D, QtCore.Qt.ShiftModifier)
+    QtWidgets.QApplication.sendEvent(plot_widget, event)
+    assert len(plot_widget.roi_list) == 0
```

Comment:
I'd also write a small one line comment for E and Shift+E telling that E removes the hoverered roi and Shift+E removes all the rois.